core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java

```diff
@@ -2349,6 +2349,27 @@ public final Sql sql(String sql) {
     assertThat(rels.get(0), isA(LogicalCalc.class));
   }
 
+  @Test void testRelShuttleForLogicalCalcShuttle() {
+    final String sql = "select count(ename) from emp";
+    final RelNode rel = tester.convertSqlToRel(sql).rel;
+    final HepProgramBuilder programBuilder = HepProgram.builder();
+    programBuilder.addRuleInstance(CoreRules.PROJECT_TO_CALC);
+    final HepPlanner planner = new HepPlanner(programBuilder.build());
+    planner.setRoot(rel);
+    final RelNode calc = planner.findBestExp();
+    final List<RelNode> rels = new ArrayList<>();
+    final RelShuttleImpl visitor = new RelShuttleImpl() {
+      @Override public RelNode visit(LogicalCalc calc) {
+        RelNode visitedRel = super.visit(calc);
+        rels.add(visitedRel);
+        return visitedRel;
+      }
+    };
+    visitor.visit(calc);
+    assertThat(rels.size(), is(1));
+    assertThat(rels.get(0), isA(LogicalCalc.class));
+  }
+
```

Comment:
You also add RelShuttle support for some other operators, like LogicalWindow, LogicalTableModify, etc. Can you also add test case for these operators?

Reply:
Thanks